core/src/scene/sceneLoader.cpp

```diff
@@ -73,21 +73,15 @@ void SceneLoader::loadShaderConfig(Node shaders, Style& style, Scene& scene) {
     if (!shaders) { return; }
 
     if (Node extensionsNode = shaders["extensions"]) {
-        std::vector<std::string> extensions;
-        if (extensionsNode.IsScalar()) {
-            extensions.push_back(extensionsNode.as<std::string>());
-        } else if(extensionsNode.IsSequence()) {
-            extensions.reserve(extensionsNode.size());
+        if(extensionsNode.IsSequence()) {
             for (const auto& extNode : extensionsNode) {
-                extensions.push_back(extNode.as<std::string>());
+                char buffer[1000]; //sufficient space
+                auto extName = extNode.as<std::string>();
+                sprintf(buffer, "#ifdef %s\n    #extension %s : enable\n    #define TANGRAM_EXTENSION_%s\n#endif",
```

Comment:
Yes, injection will happen after all mixing has been done.

wrt requirement for extensions to be at the top of the shader... I think there is no documentation saying that, but we have seen some issues in the past which were fixed when extensions were put at the top.